L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,81 @@
+Title
+----
+* Author(s): James Newton-King
+* Approver: a11r
+* Status: Draft
+* Implemented in: csharp
+* Last updated: 2021-05-19
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+grpc-dotnet currently has no load balancing support. A channel is always created for a single address.
+
+This documentation outlines an initial load balancing implementation for the grpc-dotnet client.
+
+## Background
+
+grpc-dotnet currently doesn't support any form of load balancing in the client. The changes outlined in this proposal will enable basic gRPC load balancing and some of its related features. These include:
+
+* Name resolving. Built-in support is included for resolving from DNS, and a static resolver. Custom resolvers can be created.
+* Load balancing strategies. Built-in support is included for pick first and round robin load balancing. Custom load balancers can be created.
+* Connectivity status at sub-channel and channel level
+* Wait for ready
+
+## Proposal
+
+![Diagram](L80_graphics/diagram.png)
+
+### Initialization
+
+A channel is created with an address, an optional service provider, and an optional service config.
+
+```csharp
+var channel = GrpcChannel.ForAddress(
```

Comment:
Yes, supporting this shouldn't be an issue now.

Reply:
I don't see this change.

Reply:
Added to API changes. Note that instead of `GetState(bool requestReady)`, there is a separate `ConnectAsync()` method. This matches what C# C-core does.